From 01bcfcb5a7f510254696ab04bb447364d1470425 Mon Sep 17 00:00:00 2001
From: Max Wipfli <max.wipfli@axelera.ai>
Date: Fri, 5 Jul 2024 13:54:07 +0000
Subject: [PATCH 7/9] Enable optimizations for MCU firmware

This significantly speeds up MCU startup (PLIC init, ...).

This change should not affect functionality.
---
 bin/mcu/Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/bin/mcu/Makefile b/bin/mcu/Makefile
index a7ec79c..96b5cc8 100644
--- a/bin/mcu/Makefile
+++ b/bin/mcu/Makefile
@@ -19,10 +19,10 @@ else
 endif
 
 ifeq ($(TYPE),NEWMCU)
- CFLAGS  = -O0 -g -DAL_NEWMCU=1 -Isrc
+ CFLAGS  = -O2 -g -DAL_NEWMCU=1 -Isrc
  LINKFLAGS = -Tlinker.lcf -nostartfiles
 else
- CFLAGS = -D__INT32  -O0 -g
+ CFLAGS = -D__INT32  -O2 -g
  LINKFLAGS = -lhal -lsim
 endif
 
-- 
2.43.0

